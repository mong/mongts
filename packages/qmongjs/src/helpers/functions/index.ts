export { mathClamp } from "./mathClamp";
export { validateTreatmentUnits } from "./validateTreatmentUnits";
export { filterOrderIndID } from "./filterorderindid";
export { imgLoader } from "./imgLoader";
export { level, level2 } from "./defineLevel";
export { customFormat } from "./localFormater";
export { levelSymbols, newLevelSymbols } from "./levelSymbols";
export { getLastCompleteYear } from "./getLastCompleteYear";
