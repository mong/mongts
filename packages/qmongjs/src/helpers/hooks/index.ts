export { useResizeObserver } from "./useResizeObserver";
export { useEventListener } from "./useEventListener";
export { useShouldReinitialize } from "./useShouldReinitialize";
export {
  useDescriptionQuery,
  useIndicatorQuery,
  useSelectionYearsQuery,
  useUnitNamesQuery,
  fetchRegisterNames,
  useRegisterNamesQuery,
  useMedicalFieldsQuery,
  useUnitUrlsQuery,
  useRegistryRankQuery,
  type FetchIndicatorParams,
} from "./apihooks";
export {
  elementDimensions,
  useLegendItemPosition,
  useTextWidth,
} from "./chartHooks";
