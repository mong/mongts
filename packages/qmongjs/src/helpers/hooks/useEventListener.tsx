import { useEffect, useRef } from "react";

export const useEventListener = (
  eventName: string,
  handler: (event: KeyboardEvent) => void,
  targetElement = global,
) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const listenerRef = useRef<any>();

  useEffect(() => {
    listenerRef.current = handler;
  }, [handler]);

  useEffect(() => {
    const isSupported = targetElement && targetElement.addEventListener;
    if (!isSupported) return;
    const eventListener = (event: Event) =>
      !!listenerRef.current && listenerRef.current(event);
    targetElement.addEventListener(eventName, eventListener);
    return () => {
      targetElement.removeEventListener(eventName, eventListener);
    };
  }, [eventName, targetElement]);
};

export default useEventListener;
