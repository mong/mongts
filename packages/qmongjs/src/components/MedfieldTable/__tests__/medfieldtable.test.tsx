import React from "react";
import { MedfieldTable, createMedfieldTableData } from "..";
import { vi, test, expect } from "vitest";
import * as hooks from "../../../helpers/hooks";
import { render } from "@testing-library/react";
import { vi } from "vitest";
import { Indicator } from "types";

const medfieldTableData: Indicator[] = [
  {
    // Green
    ind_id: "hjerte_ind1",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 112,
    var: 0.95,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: 10,
    level_direction: 1,
    level_green: 0.9,
    level_yellow: 0.85,
    sformat: ",.0%",
    registry_id: 1,
    registry_name: "hjertereg1",
    registry_full_name: "Hjerteregister nummer 1",
    registry_short_name: "Hjerteregister 1",
    medfield_id: 1,
    medfield_name: "hjerte",
    medfield_full_name: "Hjerte- og karsykdommer",
    ind_title: "Tittel til indikator nr. 1 for hjerte- og karsykdommer",
  },
  {
    // Red
    ind_id: "hjerte_ind2",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 15,
    var: 0.85,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: 10,
    level_direction: 1,
    level_green: 0.95,
    level_yellow: 0.9,
    sformat: ",.0%",
    registry_id: 2,
    registry_name: "hjertereg2",
    registry_full_name: "Hjerteregister nummer 2",
    registry_short_name: "Hjerteregister 2",
    medfield_id: 1,
    medfield_name: "hjerte",
    medfield_full_name: "Hjerte- og karsykdommer",
    ind_title: "Tittel til indikator nr. 2 for hjerte- og karsykdommer",
  },
  {
    // Yellow
    ind_id: "diabetes_ind1",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 54,
    var: 0.55,
    dg: null,
    type: "andel",
    include: 1,
    min_denominator: null,
    level_direction: 1,
    level_green: 0.6,
    level_yellow: 0.4,
    sformat: ",.0%",
    registry_id: 3,
    registry_name: "diabetesreg1",
    registry_full_name: "Diabetesregister nummer 1",
    registry_short_name: "Diabetesregister 1",
    medfield_id: 3,
    medfield_name: "diabetes",
    medfield_full_name: "Diabetes",
    ind_title: "Tittel til indikator nr. 1 for diabetes",
  },
  {
    // Green
    ind_id: "diabetes_ind2",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 48,
    var: 0.65,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: null,
    level_direction: 1,
    level_green: 0.6,
    level_yellow: 0.4,
    sformat: ",.0%",
    registry_id: 3,
    registry_name: "diabetesreg1",
    registry_full_name: "Diabetesregister nummer 1",
    registry_short_name: "Diabetesregister 1",
    medfield_id: 3,
    medfield_name: "diabetes",
    medfield_full_name: "Diabetes",
    ind_title: "Tittel til indikator nr. 2 for diabetes",
  },
  {
    // Green
    ind_id: "barn_ind1",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 83,
    var: 0.45,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: null,
    level_direction: 1,
    level_green: 0.4,
    level_yellow: 0.2,
    sformat: ",.0%",
    registry_id: 4,
    registry_name: "barnereg1",
    registry_full_name: "Barneregister nummer 1",
    registry_short_name: "Barneregister 1",
    medfield_id: 5,
    medfield_name: "barn",
    medfield_full_name: "Barn",
    ind_title: "Tittel til indikator nr. 1 for barn",
  },
  {
    // Green
    ind_id: "barn_ind2",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 86,
    var: 0.45,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: null,
    level_direction: 1,
    level_green: 0.4,
    level_yellow: 0.2,
    sformat: ",.0%",
    registry_id: 5,
    registry_name: "barnereg2",
    registry_full_name: "Barneregister nummer 2",
    registry_short_name: "Barneregister 2",
    medfield_id: 5,
    medfield_name: "barn",
    medfield_full_name: "Barn",
    ind_title: "Tittel til indikator nr. 2 for barn",
  },
  {
    // Green
    ind_id: "barn_ind3",
    unit_level: "hospital",
    unit_name: "Tromsø",
    context: "caregiver",
    year: 2022,
    denominator: 86,
    var: 0.45,
    dg: 0.95,
    type: "andel",
    include: 1,
    min_denominator: null,
    level_direction: 1,
    level_green: 0.4,
    level_yellow: 0.2,
    sformat: ",.0%",
    registry_id: 5,
    registry_name: "barnereg2",
    registry_full_name: "Barneregister nummer 2",
    registry_short_name: "Barneregister 2",
    medfield_id: 5,
    medfield_name: "barn",
    medfield_full_name: "Barn",
    ind_title: "Tittel til indikator nr. 3 for barn",
  },
];

vi.mock("next/font/google", () => ({
  Plus_Jakarta_Sans: () => ({
    weight: ["200", "300", "400", "500", "600", "700", "800"],
    subsets: ["latin"],
    display: "swap",
    style: {
      fontFamily: "Plus Jakarta Sans",
    },
  }),
}));

const data = createMedfieldTableData(medfieldTableData);

test("Levels counts are correct", () => {
  expect(data).toMatchSnapshot();
});

test("Table renders correctly", async () => {
  vi.spyOn(hooks, "useIndicatorQuery").mockReturnValue({
    data: medfieldTableData,
    isLoading: false,
    error: false,
  });

  const { container } = render(
    <MedfieldTable
      unitNames={["Tromsø"]}
      treatmentYear={2022}
      context={"caregiver"}
      type={"ind"}
    />,
  );

  expect(container).toMatchSnapshot();
});
