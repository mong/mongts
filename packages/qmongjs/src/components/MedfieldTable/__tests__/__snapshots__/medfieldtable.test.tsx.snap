// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Levels counts are correct 1`] = `
[
  ,
  {
    "green": 1,
    "name": "Hjerte- og karsykdommer",
    "red": 1,
    "registers": [
      ,
      {
        "full_name": "Hjerteregister nummer 1",
        "green": 1,
        "name": "hjertereg1",
        "red": 0,
        "short_name": "Hjerteregister 1",
        "yellow": 0,
      },
      {
        "full_name": "Hjerteregister nummer 2",
        "green": 0,
        "name": "hjertereg2",
        "red": 1,
        "short_name": "Hjerteregister 2",
        "yellow": 0,
      },
    ],
    "yellow": 0,
  },
  ,
  {
    "green": 1,
    "name": "Diabetes",
    "red": 0,
    "registers": [
      ,
      ,
      ,
      {
        "full_name": "Diabetesregister nummer 1",
        "green": 1,
        "name": "diabetesreg1",
        "red": 0,
        "short_name": "Diabetesregister 1",
        "yellow": 1,
      },
    ],
    "yellow": 1,
  },
  ,
  {
    "green": 3,
    "name": "Barn",
    "red": 0,
    "registers": [
      ,
      ,
      ,
      ,
      {
        "full_name": "Barneregister nummer 1",
        "green": 1,
        "name": "barnereg1",
        "red": 0,
        "short_name": "Barneregister 1",
        "yellow": 0,
      },
      {
        "full_name": "Barneregister nummer 2",
        "green": 2,
        "name": "barnereg2",
        "red": 0,
        "short_name": "Barneregister 2",
        "yellow": 0,
      },
    ],
    "yellow": 0,
  },
]
`;

exports[`Table renders correctly 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1t7lzbz-MuiTableCell-root"
            scope="col"
          />
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1t7lzbz-MuiTableCell-root"
            scope="col"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1ixrkmk-MuiTypography-root"
            >
              <b>
                Fagområde
              </b>
            </h6>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1t7lzbz-MuiTableCell-root"
            scope="col"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1ixrkmk-MuiTypography-root"
            >
              <b>
                Resultat
              </b>
            </h6>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                data-testid="ExpandCircleDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.08 9.59 12 12.67 8.92 9.59 7.5 11l4.5 4.5 4.5-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
                />
              </svg>
            </button>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
            >
              Hjerte- og karsykdommer
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <br />
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(216, 81, 64); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 464H48a16 16 0 0 1-14.07-23.62l208-384a16 16 0 0 1 28.14 0l208 384A16 16 0 0 1 464 464z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Register
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Resultat
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1apkmlw-MuiButtonBase-root-MuiButton-root"
              href="/behandlingskvalitet/hjertereg1/?selected_treatment_units=Tromsø"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiStack-root css-1rlitzi-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
                >
                  Hjerteregister 1
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1apkmlw-MuiButtonBase-root-MuiButton-root"
              href="/behandlingskvalitet/hjertereg2/?selected_treatment_units=Tromsø"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiStack-root css-1rlitzi-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
                >
                  Hjerteregister 2
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(216, 81, 64); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 464H48a16 16 0 0 1-14.07-23.62l208-384a16 16 0 0 1 28.14 0l208 384A16 16 0 0 1 464 464z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                data-testid="ExpandCircleDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.08 9.59 12 12.67 8.92 9.59 7.5 11l4.5 4.5 4.5-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
                />
              </svg>
            </button>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
            >
              Diabetes
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <br />
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(253, 156, 0); font-size: 1.2rem;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Register
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Resultat
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1apkmlw-MuiButtonBase-root-MuiButton-root"
              href="/behandlingskvalitet/diabetesreg1/?selected_treatment_units=Tromsø"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiStack-root css-1rlitzi-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
                >
                  Diabetesregister 1
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(253, 156, 0); font-size: 1.2rem;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          style="cursor: pointer;"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-hizwh9-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                data-testid="ExpandCircleDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.08 9.59 12 12.67 8.92 9.59 7.5 11l4.5 4.5 4.5-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
                />
              </svg>
            </button>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
            >
              Barn
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Register
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
            >
              Resultat
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1apkmlw-MuiButtonBase-root-MuiButton-root"
              href="/behandlingskvalitet/barnereg1/?selected_treatment_units=Tromsø"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiStack-root css-1rlitzi-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
                >
                  Barneregister 1
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-pibo8m-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1apkmlw-MuiButtonBase-root-MuiButton-root"
              href="/behandlingskvalitet/barnereg2/?selected_treatment_units=Tromsø"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiStack-root css-1rlitzi-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-140vwic-MuiTypography-root"
                >
                  Barneregister 2
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
                  data-testid="ArrowOutwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 6v2h8.59L5 17.59 6.41 19 16 9.41V18h2V6z"
                  />
                </svg>
              </div>
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-10ukr6t-MuiTableCell-root"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(88, 165, 92); font-size: 1.2rem;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
